app/file.go

```diff
@@ -1245,3 +1248,49 @@ func (a *App) CopyFileInfos(userId string, fileIds []string) ([]string, *model.A
 
 	return newFileIds, nil
 }
+
+// This function zip's up all the files in fileDatas array and then saves it to the directory specified with the specified zip file name
+// Ensure the zip file name ends with a .zip
+func (a *App) CreateZipFileAndAddFiles(fileBackend filesstore.FileBackend, fileDatas []model.FileData, zipFileName, directory string) error {
```

Comment:
Here is the utility function as we discussed 🎉 !! 
It follows what we had discussed.